frontends/web/src/routes/exchange/pocket.tsx

```diff
@@ -175,6 +176,83 @@ export const Pocket = ({ code }: TProps) => {
     });
   };
 
+  const handlePaymentRequest = useCallback(async (message: PaymentRequestV0Message) => {
+    if (!message.slip24) {
+      alertUser(t('unknownError', { errorMessage: 'Missing payment request data' }));
+      return;
+    }
+
+    // this allows to correctly handle sats mode
+    const parsedAmount = await parseExternalBtcAmount(message.amount.toString());
+    if (!parsedAmount.success) {
+      alertUser(t('unknownError', { errorMessage: 'Invalid amount' }));
+      return;
+    }
+
+    const txInput: TTxInput = {
+      address: message.bitcoinAddress,
+      amount: parsedAmount.amount,
+      feeTarget: 'high',
+      customFee: '',
+      sendAll: 'no',
+      selectedUTXOs: [],
+      paymentRequest: message.slip24,
+    };
+
+    let result = await proposeTx(code, txInput);
+    if (result.success) {
+      let txNote = t('buy.pocket.paymentRequestNote') + ' ' + message.slip24.recipientName;
+      await proposeTxNote(code, txNote);
+      sendTx(code).then((sendResult: ISendTx) => {
```

Comment:
you are in a nice async function, so please don't mix `await` and `.then` :smile: 

You can do `try { const sendResult = await sendTx(code); ... } finally { ... }`

Reply:
fixed :+1: